src/RadixDlt.NetworkGateway.DataAggregator/Services/FetchedTransactionStore.cs

```diff
@@ -161,15 +161,22 @@ public List<CoreApiSdk.Model.CommittedTransaction> GetTransactionBatch(long from
 
         if (nodeWithMostTransactions == null)
         {
-            return new List<CoreModel.CommittedTransaction>();
+            return [];
         }
 
-        return nodeWithMostTransactions
+        var transactions = nodeWithMostTransactions
             .Where(x => x.Key >= fromStateVersion)
             .OrderBy(x => x.Key)
             .Take(maxBatchSize)
             .Select(x => x.Value.CommittedTransaction)
             .ToList();
+
+        if (transactions.Count < minBatchSize)
+        {
+            return [];
```

Comment:
I fear this won't work well if there is a large min batch size -- because the node has a dynamic upper batch size dependent on how large the response is. This is to protect the Node and Gateway from an out-of-memory if the response is too large.

This code as is could get us stuck, if, `ShouldFetchNewTransactions` is false after 4 transactions, but our min batch size is 5.

Instead, we should probably also check that ShouldFetchNewTransactions is true for that node name before returning `[]` - else we should just handle the transactions as-is.

We should also comment this I imagine!

Reply:
added explicit exception informing that gateway is missconfigured and will not process transaction in such config.